[16.0][ADD] stock_release_channel_shipment_lead_time

anothingguy opened the pull request.

grindtildeath committed:
Rename module to stock_storage_type_putaway_abc

gurneyalex committed:
[FIX] crash when product is None

if product is None, then accessing the abc_storage will crash.

This happens when the method is called from
stock_storage_type_putaway_strategy/models/stock_storage_location_sequence.py, line 25

grindtildeath committed:
[IMP] stock_storage_type_putaway_abc: black, isort

grindtildeath committed:
[MIG] stock_storage_type_putaway_abc: Migration to 13.0

Allow to compute restrictions for multiple products and quants

Use helper field from stock_location_children

Remove dependency after merge of putaway_strategy into storage_type

pre-commit + setup tools

gurneyalex committed:
[FIX] issues in stock_storage_type_putaway_abc

* multiple assignment would use the same record for a_locations, b_locations and c_locations -> no sorting performed
* if product is an empty recordset, it has no abc_storage -> a useless sorting is performed
* the union operator on recordsets is not very efficient -> prefer list.append and browse the result

grindtildeath committed:
Shuffle ABC locations to select a bin randomly

Commit:
Find leaf sub-locations instead of all children

We will never do a put-away in an intermediate location,
so they should never be kept as candidate locations for the
put-away.

Follow the change in https://github.com/OCA/wms/pull/12

oca-travis committed:
[UPD] Update stock_storage_type_putaway_abc.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
stock_storage_type_putaway_abc 13.0.1.0.1

sebalix committed:
[FIX] stock_storage_type_putaway_abc: choose B locations instead of A if there is no C

sebalix committed:
[IMP] stock_storage_type_putaway_abc: sort also locations by max heights

For ABC strategy we apply the same sort among on each chunk of
locations "A", "B" and "C".

If we have the following locations:

    LOC_A_MAXHEIGHT_200
    LOC_A_MAXHEIGHT_100
    LOC_B_MAXHEIGHT_400
    LOC_B_MAXHEIGHT_300
    LOC_C_MAXHEIGHT_600
    LOC_C_MAXHEIGHT_500

If the ABC storage of a product is flagged B, then the proposed
locations will be ordered as:

    LOC_B_MAXHEIGHT_300
    LOC_B_MAXHEIGHT_400
    LOC_C_MAXHEIGHT_500
    LOC_C_MAXHEIGHT_600
    LOC_A_MAXHEIGHT_100
    LOC_A_MAXHEIGHT_200

(ordered at first with BCA as usual, then by max_height)

sebalix committed:
[FIX] stock_storage_type_putaway_abc: fix tests to compare list of IDs instead of recordset

Commit:
fix error when we have not all kinds (A/B/C) of locations

OCA-git-bot committed:
stock_storage_type_putaway_abc 13.0.1.1.0

sebastienbeau committed:
[IMP] black, isort, prettier

sebastienbeau committed:
[IMP] set module uninstallable

jbaudoux committed:
Change sorting of locations in the pack putaway computation

Invert the sorting for abc packs storage strategy to first apply the
max_height which is a physical constraint, then apply the abc
preference.

Add a pack putaway sequence to be able to favor some locations (e.g.
level). For ordered children strategy, finalize by sorting by location
name. For abc, random selection is applied at last as it was already.

Replaced multi-steps python sorting by an unique SQL quicksort query to
quickly compute the result on large stock_location table. This is
especially important for operations containing many moves requiring a
putaway computation.

sebalix committed:
[MIG] stock_storage_type_putaway_abc: Migration to 14.0

oca-travis committed:
[UPD] Update stock_storage_type_putaway_abc.pot

OCA-git-bot committed:
[UPD] README.rst

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (11 of 11 strings)

Translation: wms-14.0/wms-14.0-stock_storage_type_putaway_abc
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-stock_storage_type_putaway_abc/es_AR/

rousseldenis committed:
[IMP] stock_storage_type_putaway_abc: pre-commit stuff

rousseldenis committed:
[MIG][16.0] stock_storage_type_putaway_abc

rousseldenis committed:
[IMP] stock_storage_type_putaway_abc: Add a hook in order to allow overriding abc retrieval

rousseldenis committed:
[16.0][ADD] stock_release_channel_process_end_date

sbejaoui committed:
[ADD] stock_release_channel_shipment_advice

This module is used to plan the  shipment advice for released pickings in a
stock release channel.

As this is the base module, it provides only a simple
planning mode. All released pickings are grouped together in a single advice.

fix label

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

[FIX] - button_show_shipment_advice: improve domain

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

[IMP] - _compute_can_plan_shipment: improve expression

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

[IMP] - _plan_shipment: improve expression

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

Apply suggestions from code review

Co-authored-by: Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>

rousseldenis committed:
[IMP] stock_release_channel_process_end_time: Add timezone for end time

As time should be timezoned to correctly compute the real end date, consider
the channel warehouse timezone or company one. If not defined, consider the timezone
as UTC.

rousseldenis committed:
[FIX] stock_release_channel_process_end_time: Don't take into account seconds and microseconds

sbejaoui committed:
[IMP] s_release_channel_shipment_advice: add dock to release channel

dock and warehouse are propgated from the release channel to shipment advice planner

sbejaoui committed:
[IMP] s_release_channel_shipment_advice: check dock WH

add data validation + form behavior improvement

sbejaoui committed:
[IMP] s_release_channel_shipment_advice: update description

rousseldenis committed:
[IMP] stock_release_channel_process_end_time: Check if end time is defined before assignation to scheduled_date

lmignon committed:
[FIX] stock_release_channel_process_end_time: releasable picking computation

The compute of releasable picking must take into account the process end date of the stock release channel and not the current datetime.
The way the ORM is working into Odoo prevent to makes query on model with criteria based on values from linked model (comparing column from model A to column of model B). To work arround this limitation, we use a specialized field with a search method. Field's search methods are considered as 'internal' search methods and therefore allows you to use a special 'inselect' operator. This kind operator is usefull since it allows to express a criteria as a plain SQL quey. Thanks to this approach we've an easy way to implement our needs with the Odoo's ORM.

rousseldenis committed:
[IMP] stock_storage_type_putaway_abc: Adapt package_type_putaway_sequence field name

rousseldenis committed:
[FIX] stock_storage_putaway_abc: Use flush_model() instead of flush()

sbejaoui committed:
[FIX] s_release_channel_shipment_advice: fix method param

sbejaoui committed:
[FIX] use picking type warehouse to filter pickings to plan

lmignon committed:
[IMP] s_release_channel_shipment_advice: speedup compute

sbejaoui committed:
[FIX] s_release_channel_shipment_advice: fix compute picking_to_plan

self in compute is composed of objects browse(NewId), setting the computed field
with the db id leads to odoo raises and error saying newId didn't get a value

rousseldenis committed:
[IMP] s_release_channel_shipment_advice: tree view default

Also filter on Today by default

sbejaoui committed:
[FIX] - assign channel end date to picking after release

sbejaoui committed:
[IMP] - improve process end time location in form view

sbejaoui committed:
[FIX] fix help message & add UTC as defauly value for process_end_time_tz

Co-authored-by: Jacques-Etienne Baudoux <je@bcim.be>

jsanchezdeavila committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (73 of 73 strings)

Translation: wms-16.0/wms-16.0-stock_warehouse_flow
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_warehouse_flow/es/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: wms-16.0/wms-16.0-stock_release_channel_batch_mode_commercial_partner
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_release_channel_batch_mode_commercial_partner/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: wms-16.0/wms-16.0-stock_picking_type_shipping_policy
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_picking_type_shipping_policy/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: wms-16.0/wms-16.0-stock_release_channel_auto_release
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_release_channel_auto_release/it/

nguyenminhchien commented on stock_release_channel_shipment_lead_time/tests/__init__.py:
```suggestion
from . import test_assign_channel_shipment_date
```

nguyenminhchien commented on stock_release_channel_shipment_lead_time/tests/test_release_shipment_date.py:
test is failed
![image](https://github.com/anothingguy/wms/assets/8724793/7ecfea4c-b33c-447a-ab88-afd01e9ba311)

nguyenminhchien commented on stock_release_channel_shipment_lead_time/models/shipment_advice.py:
It's not a big issue but i think you should set `readonly=true` on the form view instead. Otherwise, you cannot import its value from excel / csv file using odoo's import feature.

anothingguy commented on stock_release_channel_shipment_lead_time/tests/__init__.py:
updated

anothingguy commented on stock_release_channel_shipment_lead_time/tests/test_release_shipment_date.py:
fixed

anothingguy commented on stock_release_channel_shipment_lead_time/models/shipment_advice.py:
I changed readonly attribute same as other fields in original shipmnet_advice module

nguyenminhchien reviewed (approved):
lgtm

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10 of 10 strings)

Translation: wms-16.0/wms-16.0-stock_picking_completion_info
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_picking_completion_info/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 95.6% (44 of 46 strings)

Translation: wms-16.0/wms-16.0-stock_dynamic_routing
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_dynamic_routing/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (73 of 73 strings)

Translation: wms-16.0/wms-16.0-stock_warehouse_flow
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_warehouse_flow/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (46 of 46 strings)

Translation: wms-16.0/wms-16.0-stock_dynamic_routing
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_dynamic_routing/es/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: wms-16.0/wms-16.0-stock_release_channel_auto_release
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_release_channel_auto_release/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (75 of 75 strings)

Translation: wms-16.0/wms-16.0-stock_available_to_promise_release
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_available_to_promise_release/it/

rousseldenis committed:
[UPD] Copier update

rousseldenis committed:
Merge pull request #752 from acsone/16.0-imp-copier-dro

[UPD] Copier update

OCA-git-bot committed:
Merge PR #496 into 16.0

Signed-off-by rousseldenis

Commit:
[UPD] Update stock_storage_type_putaway_abc.pot

OCA-git-bot committed:
oca-github-bot post-merge updates

anothingguy committed:
[IMP] add configuration to enable/disable set scheduled date

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-16.0/wms-16.0-stock_storage_type_putaway_abc
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_storage_type_putaway_abc/

anothingguy committed:
[ADD] stock_release_channel_shipment_lead_time

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: wms-16.0/wms-16.0-stock_storage_type_putaway_abc
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_storage_type_putaway_abc/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (110 of 110 strings)

Translation: wms-16.0/wms-16.0-stock_storage_type
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_storage_type/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (73 of 73 strings)

Translation: wms-16.0/wms-16.0-stock_warehouse_flow
Translate-URL: https://translation.odoo-community.org/projects/wms-16-0/wms-16-0-stock_warehouse_flow/it/

rousseldenis committed:
[IMP] stock_release_channel_process_end_time: Rename configuration parameter

nguyenminhchien committed:
[IMP] stock_storage_type: Don't rely on max_weight on product.packaging

OCA-git-bot committed:
Merge PR #760 into 16.0

Signed-off-by simahawk

OCA-git-bot committed:
[BOT] post-merge updates

simahawk committed:
s_release_channel_shipment_advice: update authorship and readme

jbaudoux committed:
s_release_channel_shipment_advice: fix form view

Move options in Options tab

OCA-git-bot committed:
Merge PR #761 into 16.0

Signed-off-by simahawk

Commit:
[UPD] Update stock_release_channel_shipment_advice.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #550 into 16.0

Signed-off-by rousseldenis

Commit:
[UPD] Update stock_release_channel_process_end_time.pot

OCA-git-bot committed:
[BOT] post-merge updates
